tests/pfcwd/test_pfcwd_function.py

```diff
@@ -0,0 +1,518 @@
+        self.pfc_wd_test_pkt_count = pfc_params['test_pkt_count']
+        self.pfc_wd_rx_port_id = pfc_params['rx_port_id']
+        self.pfc_wd_test_port =  pfc_params['test_port']
+        self.pfc_wd_test_port_id = pfc_params['test_port_id']
+        self.pfc_wd_test_port_ids = pfc_params['test_port_ids']
+        self.pfc_wd_test_neighbor_addr = pfc_params['test_neighbor_addr']
+        self.pfc_wd_rx_neighbor_addr = pfc_params['rx_neighbor_addr']
+        self.port_type = pfc_params['port_type']
+
+    def verify_tx_egress(self, action):
+        """
+        Send traffic with test port as the egress and verify if the packets get forwarded
+        or dropped based on the action
+
+        Args:
+            action(string) : PTF test action
+        """
+        logger.info("Check for egress {} on Tx port {}".format(action, self.pfc_wd_test_port))
+        ptf_params = {'router_mac': self.eth0_mac,
+                      'queue_index': self.pfc_queue_index,
+                      'pkt_count': self.pfc_wd_test_pkt_count,
+                      'port_src': self.pfc_wd_rx_port_id[0],
+                      'port_dst': "[" + str(self.pfc_wd_test_port_id) + "]",
+                      'ip_dst': self.pfc_wd_test_neighbor_addr,
+                      'port_type': self.port_type,
+                      'wd_action': action}
+        log_format = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M:%S")
+        log_file = "/tmp/pfc_wd.PfcWdTest.{}.log".format(log_format)
+        ptf_runner(self.ptf, "ptftests", "pfc_wd.PfcWdTest", "ptftests", params=ptf_params,
+                   log_file=log_file)
+
+    def verify_rx_ingress(self, action):
+        """
+        Send traffic with test port as the ingress and verify if the packets get forwarded
+        or dropped based on the action
+
+        Args:
+            action(string) : PTF test action
+        """
+        logger.info("Check for ingress {} on Rx port {}".format(action, self.pfc_wd_test_port))
+        if type(self.pfc_wd_rx_port_id) == list:
+            dst_port = "".join(str(self.pfc_wd_rx_port_id)).replace(',', '')
+        else:
+            dst_port = "[ " + str(self.pfc_wd_rx_port_id) + " ]"
+        ptf_params = {'router_mac': self.eth0_mac,
+                      'queue_index': self.pfc_queue_index,
+                      'pkt_count': self.pfc_wd_test_pkt_count,
+                      'port_src': self.pfc_wd_test_port_id,
+                      'port_dst': dst_port,
+                      'ip_dst': self.pfc_wd_rx_neighbor_addr,
+                      'port_type': self.port_type,
+                      'wd_action': action}
+        log_format = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M:%S")
+        log_file = "/tmp/pfc_wd.PfcWdTest.{}.log".format(log_format)
+        ptf_runner(self.ptf, "ptftests", "pfc_wd.PfcWdTest", "ptftests", params=ptf_params,
+                   log_file=log_file)
+
+    def verify_other_queue(self):
```

Comment:
What is the other queue? I am *guessing* based on context that this is some queue that doesn't have the watchdog running on it, but I would make it more explicit what we're comparing against here.

Reply:
will rename to verify_other_pfc_queue and add some info in the docstring